### ISSUE TITLE
Don't overwrite explicit Host for unix socket

### ISSUE THREAD

**Issue by pbrisbin** [excon/excon#630]
With #548, you made it so that `Host=""` is always sent when talking over a unix socket, discarding any `Host` I've explicitly set in the client. For what it's worth, I totally agree with your reading of the RFC:

> If the requested URI does not include an Internet host name for the service being requested, then the Host header field MUST  be given with an empty value

Unfortunately, I think there's a bug in nginx (at least in v1.12.0): it treats this the same as an omitted `Host` header and responds 400:

```
2017/06/12 08:03:51 [info] 4245#4245: *4 client sent invalid host header while reading client request headers, client: unix:, server: , request: "POST / HTTP/1.1", host: ""
unix: - - [12/Jun/2017:08:03:51 -0400] "POST / HTTP/1.1" 400 173 "-" "excon/0.56.0"
```

At least one responder on their [mailing list][ml] believes this to be correct behavior.

[ml]: http://mailman.nginx.org/pipermail/nginx-devel/2015-July/007141.html

While I wouldn't normally advocate changing one project to work around a bug in another, I think it's generally surprising to discard an actual `Host` value that I've set in the client, which is one way I could work around nginx's issue: provided the server will listen for it, setting `Host` to any valid value works just fine. So, would you consider using `||= ''` instead of `= ''` in the `UNIX` case?

As another point of reference, when talking to a unix socket via `curl`, it still requires you use an argument of `http://{something}{path}` and sends `Host={something}` with the request. This seems to be the behavior of newer curl, whereas older versions would accept `http:/{path}` and send `Host=http`. [[ref](https://superuser.com/questions/834307/can-curl-send-requests-to-sockets)]

**Comment by geemus** [comment]
Thanks for the detailed report. `||=` sounds reasonable and if I'm re-reading things accurately it shouldn't break anything while also allowing this to be worked around if needed. Do you want to throw together a PR? Might be good to test around this too (though it might be awkward to do), to ensure we keep it working in the future. Thanks!

**Comment by pbrisbin** [comment]
Thanks for the quick response. I made an attempt, with some tests: #631.

**Comment by geemus** [comment]
I believe this is now fixed via #631
